src/components/nav/nav.tsx

```diff
@@ -0,0 +1,116 @@
+ * @slot primary-content-start - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the start position of the primary nav level.
+ * @slot primary-content-center - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the center position of the primary nav level.
+ * @slot primary-content-end - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the end position of the primary nav level.
+ * @slot secondary-content-start - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the start position of the secondary nav level.
+ * @slot secondary-content-end - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the end position of the secondary nav level.
+ * @slot tertiary-content-start - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the start position of the tertiary nav level.
+ * @slot tertiary-content-end - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the end position of the tertiary nav level.
+ */
+export class CalciteNav {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+  @Element() el: HTMLCalciteNavElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * When true, display a hamburger  icon and emits `calciteNavMenuActionSelect` event on user interaction.
+   */
+  @Prop({ reflect: true }) displayMenuAction = false;
+
+  /**
+   * Specifies the label of the hamburger icon.
+   */
+  @Prop({ reflect: true }) label!: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * Emits whenever the component is selected or unselected.
+   *
+   */
+  @Event() calciteNavMenuActionSelect: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private clickHandler = () => {
+    this.calciteNavMenuActionSelect.emit();
+  };
+
+  //--------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  //--------------------------------------------------------------------------
+
+  renderMenuAction(): VNode {
+    return <calcite-action icon="hamburger" onClick={this.clickHandler} text={this.label} />;
+  }
+
+  renderNavLevel(level: "primary" | "secondary" | "tertiary"): VNode {
+    const progress = this.el.querySelector('[slot="progress"]');
```

Comment:
I think we should find a way to not have to query the dom for a slot. If the slot changes, this won't change unless another render happens. I think this is a pattern we want to try and avoid if possible.